bberg/src/lookup_builder.rs

```diff
@@ -0,0 +1,366 @@
+use crate::{
+    file_writer::BBFiles,
+    utils::{create_get_const_entities, create_get_nonconst_entities},
+};
+use ast::{
+    analyzed::{AlgebraicExpression, Analyzed, Identity, IdentityKind},
+    parsed::SelectedExpressions,
+};
+use itertools::Itertools;
+use number::FieldElement;
+
+use crate::utils::sanitize_name;
+
+#[derive(Debug)]
+/// Lookup
+///
+/// Contains the information required to produce a lookup relation
+pub struct Lookup {
+    ///  the name given to the inverse helper column
+    pub attribute: Option<String>,
+    /// The name of the counts polynomial that stores the number of times a lookup is read
+    pub counts_poly: String,
+    /// the left side of the lookup
+    pub left: LookupSide,
+    /// the right side of the lookup
+    pub right: LookupSide,
+}
+
+#[derive(Debug)]
+/// PermSide
```

Comment:
Copy-paste leftover? Probably should be "LookupSide"

Reply:
Actually good point here- they are more or less the same i can unify the type